### PR TITLE
chore: update deps

### PR DESCRIPTION
**Pull request by montezume**
* Closes #395 
* Closes #394 
* Closes #393 
* Closes #392 
* Closes #391 
* Closes #390

### CODE REVIEW

File: application-templates/starter/package.json [hunk 1]
```diff
@@ -32,6 +32,6 @@
     "@commercetools-frontend/mc-scripts": "9.0.0",
     "enzyme": "3.9.0",
     "jest": "24.1.0",
-    "react-testing-library": "5.8.0"
+    "react-testing-library": "6.0.0"
```
Comment (montezume) [inline application-templates/starter/package.json]: just removes testHooks, which we don't use

File: packages/application-components/package.json [hunk 2]
```diff
@@ -45,27 +45,27 @@
...
+    "react": "16.8.3",
+    "react-dom": "16.8.3",
     "react-intl": "2.8.0",
-    "react-testing-library": "5.8.0",
+    "react-testing-library": "6.0.0",
     "storybook-readme": "4.0.5"
   },
   "peerDependencies": {
-    "@commercetools-frontend/assets": "7.x",
+    "@commercetools-frontend/assets": "9.x",
```
Comment (montezume) [inline packages/application-components/package.json]: not sure why this was pegged to version 7...

File: packages/application-shell-connectors/package.json [hunk 3]
```diff
@@ -39,10 +39,10 @@
     "lodash-es": "4.17.11",
     "moment-timezone": "^0.5.23",
     "prop-types": "15.7.2",
-    "recompose": "0.26.0"
+    "recompose": "^0.30.0"
```
Comment (montezume) [inline packages/application-shell-connectors/package.json]: it's mostly performance optimizations, no breaking changes 🤔

File: packages/mc-scripts/package.json [hunk 4]
```diff
@@ -47,7 +47,7 @@
     "optimize-css-assets-webpack-plugin": "5.0.1",
     "pegjs-loader": "0.5.4",
     "postcss": "7.0.14",
-    "postcss-color-mod-function": "2.4.3",
+    "postcss-color-mod-function": "3.0.3",
```
Comment (montezume) [inline packages/mc-scripts/package.json]: I can't find any **real** breaking changes. Mostly just dep updates, relying on postcss 7...

File: packages/application-shell-connectors/package.json (hunk 3 above)
Comment (tdeekens) [reply packages/application-shell-connectors/package.json]: This is a bit dangerous. In the mc-fe this version breaks the Modified Products view. I never understood why. But I would recommend playing it safe and keep it as is. Maybe update in the mcfe first to ensure the version works. Whatever causs things to break...

Comment (montezume) [reply packages/application-shell-connectors/package.json]: thanks. Should we add a `throw-if-pinned-deps-changed` to app kit as well then?

Comment (tdeekens) [reply packages/application-shell-connectors/package.json]: Sure. I always wondered why it is not in yarn natively or why there isnt a pkg for this. Could just be an entry in the pkg.json called noUpdate or something 🙃

Comment (emmenko) [reply packages/application-shell-connectors/package.json]: FYI: there is an issue to work on that (#100)

As for `recompose`, I think we can completely remove it as we only use it for the display name.

I'll push a commit

Comment (emmenko) [reply packages/application-shell-connectors/package.json]: Actually I'll open a new PR, as there are a few more changes.

File: packages/application-components/package.json (hunk 2 above)
Comment (emmenko) [reply packages/application-components/package.json]: Maybe we can be a bit more loose with the assets package. Like e.g. `>=7`?

**Comment by emmenko** [comment]
> As mentioned, I'll open a PR to remove `recompose`

Here we go! #400

**Pull request merged**